### PR TITLE
Issue #37: Add ability to exclude certain files from the Packager

### LINKED ISSUE #37: Add ability to exclude certain files from the Packager
When packaging zip files, we currently get a zip from GitHub, unzip, modify all .info files, and then rezip them all into a new archive.

During this process, it would be nice to be able to remove certain blacklisted files, such as CI scripts like .travis.yml, .zenci.yml, etc.

The packager currently lives in project_github.module, in `_project_github_create_package()`.

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop-contrib/project/issues/37

### CODE REVIEW

File: project_github/project_github.api.php [hunk 1]
```diff
@@ -70,3 +70,25 @@ function hook_github_project_validate_release(Node $release_node, array &$errors
...
+ * @return NULL
+ *   No return value.
+ */
+function hook_project_github_create_package_alter($project_name, array $files) {
+  $project_node = node_load($release_node->project['release_nid']);
+  if ($project_node->type === 'project_module') {
+    $release_node->type = 'module_release';
+  }
+  else {
+    $errors['project_type_invalid'] = t('Project releases are only supported on module projects.');
```
Comment (quicksketch) [inline project_github/project_github.api.php]: Looks like in this example, the `$errors` variable would not be returned anywhere.

File: project_github/project_github.api.php [hunk 2]
```diff
@@ -70,3 +70,25 @@ function hook_github_project_validate_release(Node $release_node, array &$errors
...
+ * @param $project_name
+ *   The name of the project.
+ * @param array $files
+ *   An array representing the files in the payload directory after the payload
+ *   has been unzipped, and the original zip file deleted, and before the new
+ *   archive is created and resent to GitHub.
+ * @return NULL
+ *   No return value.
+ */
+function hook_project_github_create_package_alter($project_name, array $files) {
```
Comment (quicksketch) [inline project_github/project_github.api.php]:
```suggestion
function hook_project_github_create_package_alter(array &$files, $project_name) {
```

I think $files has to be passed by reference in order for this to work.

Comment (quicksketch) [reply project_github/project_github.api.php]: Hm, and actually most alter hooks have the array that is altered as the first parameter. e.g. hook_form_alter() passes $form, $form_state, and $form_id. This is similar where the $form_id is passed last even though it is the identifier.

File: project_github/project_github.api.php (hunk 1 above)
Comment (docwilmot) [reply project_github/project_github.api.php]: Ooops, That was a copy/paste from the api example just above it. Pushed a more appropriate example

File: project_github/project_github.api.php (hunk 2 above)
Comment (docwilmot) [reply project_github/project_github.api.php]: Fixed in new commit.

**Pull request merged**